packages/@react-spectrum/overlays/src/Popover.tsx

```diff
@@ -102,6 +102,38 @@ const PopoverWrapper = forwardRef((props: PopoverWrapperProps, ref: RefObject<HT
   let {modalProps} = useModal({
     isDisabled: isNonModal
   });
+  let [isHideArrow, setIsHideArrow] = useState(hideArrow);
+
+  useLayoutEffect(() => {
+    // check position of arrow and popover boundries
+    if (ref.current && arrowProps) {
+      let propsHideArrow = hideArrow;
+
+      // figure out the location of the offset and which popover dimension to use
+      let offset = arrowProps.style.left ? 'left' : 'top';
+      let overlayCrossSize = offset === 'left' ? ref.current.offsetWidth : ref.current.offsetHeight;
+
+      // for the start and top, no change is made to arrowPosition not caught in a condiational below
+      if (12 > arrowProps.style[offset] && arrowProps.style[offset] > 0) {
+        // using the default padding for proper arrow placement
+        let arrowStyle = {...arrowProps.style};
+        arrowStyle[offset] = 12;
+        arrowProps.style = arrowStyle;
+      } else if (arrowProps.style[offset] > overlayCrossSize - 12 && arrowProps.style[offset] + 2 <= overlayCrossSize) {
+        // for the end and bottom, keep the arrow pointing at the button and within the popover
+        let arrowStyle = {...arrowProps.style};
+        arrowStyle[offset] = overlayCrossSize - 12;
+        arrowProps.style = arrowStyle;
+      } else if (arrowProps.style[offset] <= 0 || arrowProps.style[offset] + 2 > overlayCrossSize) {
+        // trigger is too far off the page, hiding the arrow per Spectrum
+        propsHideArrow = true;
+      }
```

Comment:
Should this also adjust the spacing between the Popover and the trigger?
<img width="537" alt="image" src="https://user-images.githubusercontent.com/9661127/175750529-02600bfe-0a60-4153-9759-38a1759efcc2.png">

Reply:
If I'm understanding your comment, the answer is no. We can move the popover to within 6px of the edge, from 12px, and hide the arrow. We are adjusting spacing, but hiding the arrow isn't what adjusts the spacing and we can't adjust it more then we do without talking to Spectrum again since they set these rules.

Reply:
or do you mean the offset since the arrow is gone?

Reply:
Removing any margin in the hide arrow case would move it closer to the trigger which look better.

Reply:
Yep, looks better now, I'll resolve